sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyDetection/SuppressCondition.cs

```diff
@@ -29,13 +29,19 @@ public SuppressCondition(int minimumNumber, double minimumRatio)
         /// Must include the most recent point, so value must be at least 1.
         /// </summary>
         [CodeGenMember("MinNumber")]
-        public int MinimumNumber { get; }
+        public int MinimumNumber { get; set; }
 
         /// <summary>
         /// The minimum percentage of unexpected values that must be present in the latest ingested
         /// data points to detect an anomaly. Value is between (0, 100].
         /// </summary>
         [CodeGenMember("MinRatio")]
-        public double MinimumRatio { get; }
+        public double MinimumRatio { get; set; }
+
+        internal SuppressConditionPatch GetPatchModel() => new SuppressConditionPatch()
```

Comment:
also curious because I haven't seen this, so this means that for a type that is used as input and output, if we do a rename on top of codegen, we will always have to do a transformation for it because it serializes with the name of the property we customized, instead of the property the service set?

Reply:
I'm not sure if I'm following. This transformation has no relation with CodeGen renames. The problem is that the service uses different types for Update operations. Using `DataFeed` as an example, the service:
- Accepts a `DataFeed` for the create method.
- Accepts a `DataFeedPatch` for the update method.
- Returns a `DataFeed` from the get method.
- Returns a list of `DataFeed` from the list method.

They need to use a different model for Update because:
- `DataFeedPatch` has less properties than `DataFeed`. Some properties can only be set during creation, so they won't allow users to set them in a patch model.
- `DataFeedPatch` has no required properties. The service does not force users to set all properties again during update, but some properties are mandatory during creation. So the two models have a different set of required properties.

However, we expose them as a single `DataFeed` model in the SDK. We need to do a conversion to the patch model internally for the Update method.

Reply:
that clarifies it then :) thank you!